packages/hdwallet-portis/src/portis.ts

```diff
@@ -0,0 +1,386 @@
+      await this.portisCallInProgress
+      for (let i = 0; i < msg.length; i++) {
+        const { addressNList } = msg[i];
+        const portisResult = await this.portis.getExtendedPublicKey(addressNListToBIP32(addressNList))
+        const { result, error } = portisResult
+        if(error) {
+          console.log('getPublicKeys error', error)
+        }
+        publicKeys.push({ xpub: result })
+      }
+      resolve(publicKeys)
+    })
+    return this.portisCallInProgress
+  }
+
+  public async ethSignTx (msg: ETHSignTx): Promise<ETHSignedTx> {
+    const result = await this.web3.eth.signTransaction({
+      from: await this._ethGetAddress(),
+      to: msg.to,
+      value: msg.value,
+      gas: msg.gasLimit,
+      gasPrice: msg.gasPrice,
+      data: msg.data,
+      nonce: msg.nonce
+    })
+    return {
+        v: result.tx.v,
+        r: result.tx.r,
+        s:  result.tx.s,
+        serialized: result.raw
+    } 
+  }
+
+  public async ethSignMessage (msg: ETHSignMessage): Promise<ETHSignedMessage> {
+
+    const address = await this._ethGetAddress()
+    const result = await this.web3.eth.sign(msg.message, address)
+    return {
+      address,
+      signature: result
+    }
+  }
+
+  public ethGetAccountPaths (msg: ETHGetAccountPath): Array<ETHAccountPath> {
+    return [{
+      addressNList: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx, 0, 0 ],
+      hardenedPath: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx ],
+      relPath: [ 0, 0 ],
+      description: "Portis"
+    }]
+  }
+
+  public async ethGetAddress (msg: ETHGetAddress): Promise<string> {
+    return this._ethGetAddress()
+  }
+
+  public async getDeviceID(): Promise<string> {
+    return 'portis:' + (await this._ethGetAddress())
+  }
+
+  private async _ethGetAddress(): Promise<string> {
+    return (await this.web3.eth.getAccounts())[0]
+  }
+
+  public async getFirmwareVersion(): Promise<string> {
+    return 'portis'
+  }
+}
+
+export class PortisHDWalletInfo implements HDWalletInfo, ETHWalletInfo {
+  _supportsBTCInfo: boolean = false
+  _supportsETHInfo: boolean = true
+
+  public getVendor (): string {
+    return "Portis"
+  }
+
+  public async ethSupportsNetwork (chainId: number = 1): Promise<boolean> {
+    return chainId === 1
+  }
+
+  public async ethSupportsSecureTransfer (): Promise<boolean> {
+    return false
+  }
+
+  public async ethSupportsNativeShapeShift (): Promise<boolean> {
+    return false
+  }
+
+  public ethGetAccountPaths (msg: ETHGetAccountPath): Array<ETHAccountPath> {
+    return [{
+      addressNList: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx, 0, 0 ],
+      hardenedPath: [ 0x80000000 + 44, 0x80000000 + slip44ByCoin(msg.coin), 0x80000000 + msg.accountIdx ],
+      relPath: [ 0, 0 ],
+      description: "Portis"
+    }]
+  }
+
+  public async hasOnDevicePinEntry (): Promise<boolean> {
+    return false
```

Comment:
This feature test means "should I expect to get & handle pin requests in the client UI? or does the device handle that for us?". Since Portis doesn't have PINs, the client doesn't have to deal with those, hence this should return `true`:

```suggestion
    return true
```